packages/gestalt-codemods/next/dropdown-link.js

```diff
@@ -0,0 +1,97 @@
+/**
+  - Removes `onSelect` prop from Dropdown
+  - Renames `handleSelect` prop to `onSelect` for Dropdown.Item
+  - Splits out Dropdown.Item to include Dropdown.Link, ensures each contains correct props
+ */
+
+// yarn codemod --parser=flow -t=packages/gestalt-codemods/next/dropdown-link.js relative/path/to/your/code
+
+const commonProps = ['badgeText', 'children', 'option'];
+const dropdownItemProps = [...commonProps, 'onSelect', 'selected'];
+const dropdownLinkProps = [...commonProps, 'href', 'isExternal', 'onClick'];
+
+export default function transformer(file, api) {
+  const j = api.jscodeshift;
+  const src = j(file.source);
+  let localIdentifierNames;
+
+  src.find(j.ImportDeclaration).forEach((path) => {
+    const decl = path.node;
+    if (decl.source.value !== 'gestalt') {
+      return null;
+    }
+
+    localIdentifierNames = decl.specifiers
+      .filter((node) => node.imported?.name === 'Dropdown')
+      .map((node) => node.local?.name);
+
+    return null;
+  });
+
+  // No Dropdowns here, move along
+  if (!localIdentifierNames || localIdentifierNames.length === 0) {
+    return null;
+  }
+
+  return src
+    .find(j.JSXElement)
+    .forEach((jsxElement) => {
+      const { node } = jsxElement;
+      const nodeName = node.openingElement.name.name;
+      const nodeObjectName = node.openingElement.name.object?.name;
+      // Not used, but this is where the "Item" of "Dropdown.Item" lives
+      // const nodePropertyName = node.openingElement.name.property?.name;
```

Comment:
not really important for this codemod, but we pattern match off previous codemods, so a note to Future Us